File: min_transform2.js
function calMinTransform(n) {
  const binStr = (n >>> 0).toString(2)
  let res = 0
  let flag = true
  for (let i in binStr) {
    if (binStr[i] === '1') {
      const x = 2 ** (binStr.length - i) - 1
      if (flag) {
        console.log(x)
        res += x
        flag = false
      } else {
        flag = true

        res -= x
      }
    }
  }
  return res
}

const n = 333
const ret = calMinTransform(n)

File: min_transform2.go
package main

import (
	"fmt"
	"strings"
)

func calMinTransform(n int) int {
	binStr := fmt.Sprintf("%b", n)
	binArr := strings.Split(binStr, "")

	res := 0
	state := true
	lenBinArr := len(binArr)
	
	for i, c := range binArr {
		if c == "1" {
			steps := 1<<(lenBinArr-i) - 1
			if state {
				state = false
				res += steps
			} else {
				state = true
				res -= steps
			}
		}
	}

	return res
}

func main() {
	n := 333
	res := calMinTransform(n)
	fmt.Printf("res::%v\n", res)
}
